moz_security/sandboxes/heka/analysis/moz_security_auth_lastx.lua

```diff
@@ -72,20 +83,51 @@ local track_fields      = read_config("track_fields") or error("track_fields mus
 local geocity_field     = read_config("geocity_field")
 local geocountry_field  = read_config("geocountry_field")
 local cephost           = read_config("Hostname") or "unknown"
+local expireolderthan   = read_config("expireolderthan") or 864000
+
+_PRESERVATION_VERSION = read_config("preservation_version") or 0
 
 -- global, we want to serialize this to reload on startup
 userdata = {}
 
-local msg = {
-    Type = "alert",
-    Payload = "",
-    Severity = 1,
-    Fields = {
-        { name = "id", value = "auth_lastx" },
-        { name = "summary", value = "" },
-        { name = "email.recipients", value = {}}
+
+function get_msg(subject, defrecip, userrecip, defirc, payload)
+    local msg = {
+        Type = "alert",
+        Payload = payload,
+        Severity = 1,
+        Fields = {
+            { name = "id", value = "auth_lastx" },
+            { name = "summary", value = subject },
+        }
     }
-}
+    local i = 3
+    if defrecip then
+        if userrecip then
+            msg.Fields[i] = {name = "email.recipients", value = {defrecip, userrecip}}
+        else
+            msg.Fields[i] = {name = "email.recipients", value = {defrecip}}
+        end
+        i = i + 1
+    end
+    if defirc then
+        msg.Fields[i] = {name = "irc.target", value = defirc}
+    end
+    return msg
+end
+
+
+function prune_userdata(user, cutoff)
+    if not userdata[user] then return 0 end
```

Comment:
The user has already been verified and we have already calculated the length.  We can do the pruning, check, and oldest identification in a single pass in the main loop without adding much complexity (just the cutoff check below)